packages/quantic/force-app/main/default/lwc/quanticFeedbackModalQna/__tests__/quanticFeedbackModalQna.test.js

```diff
@@ -0,0 +1,299 @@
+};
+
+const defaultOptions = {
+  liked: false,
+  handleSubmit: functionsMocks.handleSubmit,
+};
+
+function createTestComponent(options = defaultOptions) {
+  const element = createElement('c-quantic-feedback-modal-qna', {
+    is: QuanticFeedbackModalQna,
+  });
+
+  for (const [key, value] of Object.entries(options)) {
+    element[key] = value;
+  }
+
+  document.body.appendChild(element);
+  return element;
+}
+
+// Helper function to wait until the microtask queue is empty.
+function flushPromises() {
+  return new Promise((resolve) => setTimeout(resolve, 0));
+}
+
+function lightningElementValueChange(element, selector, value) {
+  const lightningElement = element.shadowRoot.querySelector(selector);
+  // All base components in Jest tests are stubs. So we have to set the value manually.
+  // Including to the checkValidity method.
+  lightningElement.value = value;
+  lightningElement.validity = true;
+  lightningElement.checkValidity = () => true;
+  const event = new CustomEvent('change');
+  expect(lightningElement).not.toBeNull();
+  lightningElement.dispatchEvent(event);
+}
+
+describe('c-quantic-feedback-modal-qna', () => {
+  afterEach(() => {
+    // The jsdom instance is shared across test cases in a single file so reset the DOM
+    while (document.body.firstChild) {
+      document.body.removeChild(document.body.firstChild);
+    }
+    jest.clearAllMocks();
+  });
+
+  it('loads correctly', async () => {
+    const element = createTestComponent();
+    await flushPromises();
+
+    const answerEvaluationQuestions = element.shadowRoot.querySelectorAll(
+      selectors.answerEvaluationQuestions
+    );
+    expect(answerEvaluationQuestions.length).toBe(4);
+
+    const questionTopicRadio = element.shadowRoot.querySelector(
+      selectors.questionTopicRadio
+    );
+    expect(questionTopicRadio).not.toBeNull();
+
+    const questionHallucinationRadio = element.shadowRoot.querySelector(
+      selectors.questionHallucinationRadio
+    );
+    expect(questionHallucinationRadio).not.toBeNull();
+
+    const questionDocumentedRadio = element.shadowRoot.querySelector(
+      selectors.questionDocumentedRadio
+    );
+    expect(questionDocumentedRadio).not.toBeNull();
+
+    const questionReadableRadio = element.shadowRoot.querySelector(
+      selectors.questionReadableRadio
+    );
+    expect(questionReadableRadio).not.toBeNull();
+
+    const documentUrlInput = element.shadowRoot.querySelector(
+      selectors.documentUrlInput
+    );
+    expect(documentUrlInput).not.toBeNull();
+
+    const additionalDetailsInput = element.shadowRoot.querySelector(
+      selectors.additionalDetailsTextarea
+    );
+    expect(additionalDetailsInput).not.toBeNull();
+
+    const skipButton = element.shadowRoot.querySelector(selectors.skipButton);
+    expect(skipButton).not.toBeNull();
+
+    const submitButton = element.shadowRoot.querySelector(
+      selectors.submitButton
+    );
+    expect(submitButton).not.toBeNull();
```

Comment:
I would use an array of selectors here to just check with a forEach()

Reply:
No, because the tests if an element fail to render, would fail inside the loop and we would have no idea which element didn't render correctly.

If we keep it like this, we know exactly which expect failed and which element it was.

In tests, you want to be explicit even if it means writing more code.

Reply:
Are you sure? Im pretty sure I tried it somewhere before and despite the loop it was still logging in error the component in the array that was faulty...

did that in the `hostedSearchPageRenderer.spec.ts`

<img width="732" alt="image" src="https://github.com/user-attachments/assets/26686099-c2d0-4bf4-a4d5-0f321d066fd2">

Reply:
ah because you put the `it` in the loop, your example creates 1 test per component.

My test tests all "default" components in 1 test, I don't want to have 1 test per component, otherwise it will re-create and render the component X times, instead of loading once and checking for the components to be there.